coderd/httpmw/apikey.go

```diff
@@ -214,6 +214,26 @@ func ExtractAPIKey(rw http.ResponseWriter, r *http.Request, cfg ExtractAPIKeyCon
 			return nil, nil, false
 		}
 
+		// Add WWW-Authenticate header for 401/403 responses (RFC 6750)
+		if code == http.StatusUnauthorized || code == http.StatusForbidden {
+			// Basic Bearer challenge with realm
+			wwwAuth := `Bearer realm="coder"`
+
+			// Add error details based on the type of error
+			switch {
+			case strings.Contains(response.Message, "invalid") || strings.Contains(response.Detail, "invalid"):
+				wwwAuth = `Bearer realm="coder", error="invalid_token", error_description="The access token is invalid"`
+			case strings.Contains(response.Message, "expired") || strings.Contains(response.Detail, "expired"):
+				wwwAuth = `Bearer realm="coder", error="invalid_token", error_description="The access token has expired"`
+			case strings.Contains(response.Message, "audience") || strings.Contains(response.Message, "mismatch"):
+				wwwAuth = `Bearer realm="coder", error="invalid_token", error_description="The access token audience does not match this resource"`
+			case code == http.StatusForbidden:
+				wwwAuth = `Bearer realm="coder", error="insufficient_scope", error_description="The request requires higher privileges than provided by the access token"`
+			}
```

Comment:
How about we map 401 to `invalid_token` and `403` to `insufficient_scope` instead? This is what the RFC recommends.

Reply:
Yeah, I'll refactor this to switch on the status code first and then do the strings contains matching.